### PR TITLE
Madhura_Assignment_22_55mins

### PR DESCRIPTION
**Pull request by MadhuraMJun20**
Number is Palindrome or not : 
PalindromeNumber
String is Palindrome or not : 
PalindromeString

### CODE REVIEW

File: src/madhuraMJun20/scannerClass/PalindromeNumber.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+/*Verify given number is palindrome or not. 
+Input: number = 45654
+Output : Given number 45654 is palindrome.
+
+Input: number = 56789
+Output : Given number 56789 is not palindrome.
+*/
+public class PalindromeNumber {
+	int number;
```
Comment (HarshadBJun20) [inline src/madhuraMJun20/scannerClass/PalindromeNumber.java]: No need of this instance variable "number".

File: src/madhuraMJun20/scannerClass/PalindromeNumber.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	int reverseNum;
+
+	int findPalindromeNum(int number) {
+		int mod = 0;
+		while (number > 0) {
+			mod = number % 10;
+			number = number / 10;
+			reverseNum = mod + reverseNum * 10;
+		}
+		return reverseNum;
```
Comment (HarshadBJun20) [inline src/madhuraMJun20/scannerClass/PalindromeNumber.java]: No need to return reverseNum value. if you have already declared the reverseNum as instance variable. I suggest to make these changes.

File: src/madhuraMJun20/scannerClass/PalindromeString.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+			newString += String.valueOf(ch);
+		}
+		if (input.equals(newString))
+			System.out.println("Given string " + input + " is a palindrome.");
+		else
+			System.out.println("Given string " + input + " is not a palindrome.");
+	}
+
+	public static void main(String[] args) {
+		PalindromeString palindromeString = new PalindromeString();
```
Comment (HarshadBJun20) [inline src/madhuraMJun20/scannerClass/PalindromeString.java:20]: Code looks good to me.

**Review by HarshadBJun20 — CHANGES_REQUESTED** [review]
Request you to make the suggested changes.

File: src/madhuraMJun20/scannerClass/PalindromeString.java (hunk 3 above)
Comment (MadhuraMJun20) [reply src/madhuraMJun20/scannerClass/PalindromeString.java:25]: Thanks

Comment (MadhuraMJun20) [reply src/madhuraMJun20/scannerClass/PalindromeString.java:20]: Thanks

File: src/madhuraMJun20/scannerClass/PalindromeNumber.java (hunk 2 above)
Comment (MadhuraMJun20) [reply src/madhuraMJun20/scannerClass/PalindromeNumber.java]: Ok thanks. Updated the code

File: src/madhuraMJun20/scannerClass/PalindromeNumber.java (hunk 1 above)
Comment (MadhuraMJun20) [reply src/madhuraMJun20/scannerClass/PalindromeNumber.java]: Done

**Review by HarshadBJun20 — APPROVED** [review]
Hi Madhura,
Now Code looks good to me.

File: src/madhuraMJun20/scannerClass/PalindromeNumber.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package madhuraMJun20.scannerClass;
+
+/*Verify given number is palindrome or not. 
+Input: number = 45654
+Output : Given number 45654 is palindrome.*/
+public class PalindromeNumber {
+	int reverseNum;
+
+	void findPalindromeNum(int number) {
+		int mod = 0;
```
Comment (KrishnaKTechnocredits) [inline src/madhuraMJun20/scannerClass/PalindromeNumber.java:10]: No need of this variable mod.

File: src/madhuraMJun20/scannerClass/PalindromeNumber.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+		while (number > 0) {
+			mod = number % 10;      // mod= mod *10 + number% 10; number = number/10;
+			number = number / 10;
+			reverseNum = mod + reverseNum * 10;
+		}
```
Comment (KrishnaKTechnocredits) [inline src/madhuraMJun20/scannerClass/PalindromeNumber.java:15]: reverseNum = (number % 10 )+ (reverseNum * 10);
number = number / 10;

File: src/madhuraMJun20/scannerClass/PalindromeNumber.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+	}
+
+	void display(int number) {
```
Comment (KrishnaKTechnocredits) [inline src/madhuraMJun20/scannerClass/PalindromeNumber.java:18]: Why you have created extra method when you have already given method name as findPalindromeNum(num).
As per method name it looks like after calling this method I got the answer.

File: src/madhuraMJun20/scannerClass/PalindromeNumber.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+		findPalindromeNum(number);
+		if (number == reverseNum)
+			System.out.println("Given Number " + number + " is palindrome");
+		else
+			System.out.println("Given Number " + number + " is not a palindrome");
+	}
+
+	public static void main(String[] args) {
+		PalindromeNumber palindromeNumber = new PalindromeNumber();
+		palindromeNumber.display(45654);
```
Comment (KrishnaKTechnocredits) [inline src/madhuraMJun20/scannerClass/PalindromeNumber.java:28]: How anybody knows that your display is internally calling palindrome method?

File: src/madhuraMJun20/scannerClass/PalindromeString.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
+package madhuraMJun20.scannerClass;
+
+/*Verify given String is palindrome or not. 
+Input: radar
+Output : Given string radar is palindrome.*/
+public class PalindromeString {
+
+	void findPalindrome(String input) {
+		String newString = "";
+		for (int i = input.length() - 1; i >= 0; i--) {
```
Comment (KrishnaKTechnocredits) [inline src/madhuraMJun20/scannerClass/PalindromeString.java:10]: Why are you used i.  ?

**Pull request merged**